docs/developer/branching-and-workflow.md

```diff
@@ -0,0 +1,31 @@
+# Branching and Workflow
+
```

Comment:
I think this paragraph could be shorten to:

Our goal is to have minimal amount of branches in the repo: the main branch, release branches and long-term feature branches.

Reply:
I rephrased your suggestion a bit. Let me know what you think